### PR TITLE
Ranking evaluation documentation

### PR DESCRIPTION
**Pull request by bilbof**
This documents our efforts to find a 'data based method for judging the performance of changes to the search algorithm' without requiring AB tests.

Trello: https://trello.com/c/OwuWu73I

### CODE REVIEW

File: doc/rank_evaluation.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+See the Search Relevancy grafana dashboard.
+
+## How do we collect relevancy judgements?
+
+We're working on that at the moment. So far we've manually generated them.
+
+In future we could gather more internally in the organisation, or use end-user
+click data as a signal for relevancy judgements.
```
Comment (MatMoore) [inline doc/rank_evaluation.md:131]: Hi @bilbof! I investigated this for my data science accelerator project a couple of years ago and have some code on my github for doing this. I don't know if the team still has the documentation from it, but feel free to hit me up on x-gov slack if you are looking at this again and have any questions. 😄 

It's very cool to see GOV.UK experimenting with search again!

File: doc/rank_evaluation.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+By using DCG we make two assumptions:
+
+> - Highly relevant documents are more useful when appearing earlier in a search engine result list (have higher ranks)
+> - Highly relevant documents are more useful than marginally relevant documents, which are in turn more useful than non-relevant documents.
+
+DCG requires a query and a list of relevancy judgements (rated documents).
+
+We manually give a rating between 0 and 4 to documents in search results.
+4 = Totally relevant, 0 being not relevant.
```
Comment (barrucadu) [inline doc/rank_evaluation.md]: It's actually 0 to 3 in our current judgements:

- 0 = irrelevant (which is equivalent to unrated in DCG)
- 1 = misplaced
- 2 = near
- 3 = relevant

Comment (bilbof) [reply doc/rank_evaluation.md]: Thanks @barrucadu, I added this to the doc!

File: doc/rank_evaluation.md (hunk 1 above)
Comment (bilbof) [reply doc/rank_evaluation.md:131]: Hi @MatMoore! That would be fantastic, thanks! I'll send you a message. 😄

File: doc/rank_evaluation.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+debug:rank_evaluation[my-local-relevancy-judgements.csv]
+```
+
+We also report the rank evaluation scores to graphite.
+This enables us to plot how relevancy changes over time (overall
+  and for a given query).
+
+This runs every 3 hours in all environments:
+```
+SEND_TO_GRAPHITE=true debug:rank_evaluation
```
Comment (Tetrino) [inline doc/rank_evaluation.md:121]: Minor thing, but now you've swapped this elsewhere should it be swapped here?

Comment (bilbof) [reply doc/rank_evaluation.md:121]: I think the order doesn't matter. I reordered them for clarity in puppet - it was the quotes that were causing the issue there.

**Review by Tetrino — APPROVED** [review]

**Pull request merged**